PWGHF/HFC/Tasks/taskFlow.cxx

```diff
@@ -271,8 +527,41 @@ struct HfTaskFlow {
     registry.fill(HIST("Data/TpcTpc/HadronHadron/MixedEvent/hNtracksMixing"), Ntracks);
   }
 
+  // ---- MC : TPC-TPC h-h Mixed Event QA ----
+  template <typename TTracks>
+  void fillTpcTpcChChMixedEventQAmc(float multiplicity, float vz, TTracks const& tracks)
+  {
+    if constexpr (std::is_same_v<TracksWDcaSelMC, TTracks>) { // if MC Rec
+      registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hMultiplicityMixing"), multiplicity);
+      registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hVtxZMixing"), vz);
+    } else { // if MC Gen
+      registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hMultiplicityMixing"), multiplicity);
+      registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hVtxZMixing"), vz);
+    }
+
+    int Ntracks = 0;
+    for (const auto& track1 : tracks) {
+      Ntracks++;
+      if constexpr (std::is_same_v<TracksWDcaSelMC, TTracks>) { // if MC Rec
+        registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hPtMixing"), track1.pt());
+        registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hEtaMixing"), track1.eta());
+        registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hPhiMixing"), track1.phi());
+      } else { // if MC Gen
+        registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hPtMixing"), track1.pt());
+        registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hEtaMixing"), track1.eta());
+        registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hPhiMixing"), track1.phi());
+      }
+    }
+    if constexpr (std::is_same_v<TracksWDcaSelMC, TTracks>) { // if MC Rec
+      registry.fill(HIST("MC/Rec/TpcTpc/HadronHadron/MixedEvent/hNtracksMixing"), Ntracks);
+    } else { // if MC Gen
+      registry.fill(HIST("MC/Gen/TpcTpc/HadronHadron/MixedEvent/hNtracksMixing"), Ntracks);
+    }
```

Comment:
How does this guarantee that the `else` case is MC gen?

Reply:
It is within the MC QA function, and only MC Rec uses "TracksWDcaSelMC" as TTracks, while MC Gen uses a different type "aodMcParticles"